src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -131,28 +174,65 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string paramete
                     this.logger?.LogDebug("Running PowerShell rule engine");
                     evaluations = evaluations.Concat(this.powerShellRuleEngine.AnalyzeTemplate(templateContext));
                 }
+               
+                // Recursively handle nested templates 
```

Comment:
Since none of this relies on doing any analysis first, all of this should be doable inside ArmTemplateProcessor.cs.  Handling all this logic there makes more sense, because this isn't anlysis code, it's processing code.

Reply:
Theoretically, you could replace the nested template strings with the processed templates, then simply recursively analyze each pre-processed template here.

Reply:
I think we briefly considered this option then creating multiple contexts, but the fully recursive approach seemed to be an easier approach with handling different scopes and passing parameters down because you would have to process parent templates to get values to pass down to child templates.

Reply:
This may need some refactoring to do right, which may be too large to fit in the scope of this change.